### PR TITLE
CSHARP-5095: Generate ssdlc_compliance_report.md

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: evergreen/template_ssdlc_compliance_report.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# ${PRODUCT_NAME} SSDLC compliance report
```
Comment (BorisDog) [inline evergreen/template_ssdlc_compliance_report.md:1]: [JAVA template for reference](https://github.com/mongodb/mongo-java-driver/pull/1405/files#diff-1b9422c730f5e085be6b818a2769a908106fb852da82774a59a12d8b78382b7e)

File: evergreen/evergreen.yml [hunk 2]
```diff
@@ -1819,11 +1842,21 @@ tasks:
...
+        - func: download-packages
+        - func: trace-artifacts
+        - func: generate-ssdlc-report
+
     - name: validate-apidocs
       commands:
         - func: install-dotnet
         - func: build-apidocs
 
+    - name: generate-ssdlc-report
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:1856]: Do we need this separate task?

**Review by sanych-sun — APPROVED** [review]
LGTM

File: evergreen/evergreen.yml (hunk 2 above)
Comment (BorisDog) [reply evergreen/evergreen.yml:1856]: No, good catch.

File: evergreen/evergreen.yml [hunk 3]
```diff
@@ -2663,3 +2696,14 @@ buildvariants:
...
         ## add dependency onto packages smoke test once it implemented
+
+- matrix_name: ssdlc-reports
+  matrix_spec:
+    os: "ubuntu-2004"
+  display_name: "SSDLC Reports"
+  tags: ["release-tag"]
+  tasks:
+    - name: generate-ssdlc-reports
+      depends_on:
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:2707]: Please add before merge:
`git_tag_only: true`
So this variant will not be visible on variant/tasks screen

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**